exporter/otlphttpexporter/factory.go

```diff
@@ -67,6 +84,12 @@ func createTraceExporter(
 		return nil, err
 	}
 	oCfg := cfg.(*Config)
+
+	oce.tracesURL, err = composeSignalURL(oCfg, oCfg.TracesEndpoint, "traces")
```

Comment:
It is not intuitive, and does not respect the bounds of the classes. I would suggest to have this logic in the exporter not in the factory.

Reply:
@tigrannajaryan please do this in a followup PR.